### PR TITLE
ref: Convert CurrentDate to Swift

### PR DESCRIPTION
**Pull request by philipphofmann**
Using CurrentDate in Swift code is essential. Instead of exposing the existing ObjC class to Swift, I converted the class to Swift.

This PR is based on https://github.com/getsentry/sentry-cocoa/pull/3716 and is required for DDM.

#skip-changelog

### CODE REVIEW

File: Sources/Swift/Helper/SentryCurrentDateProvider.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@_implementationOnly import _SentryPrivate
```
Comment (brustolin) [inline Sources/Swift/Helper/SentryCurrentDateProvider.swift:1]: l: We dont need _SentryPrivate here
```suggestion
```

File: Sources/Swift/Helper/SentryCurrentDateProvider.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@objcMembers class SentryCurrentDateProvider: NSObject {
+
+    func date() -> Date {
+        return Date()
+    }
+
+    func dispatchTimeNow() -> DispatchTime {
+        return .now()
+    }
```
Comment (brustolin) [inline Sources/Swift/Helper/SentryCurrentDateProvider.swift:11]: l: Are we really not using this? At least we dont touch it during tests.
If so I would remove it.

**Review by brustolin — APPROVED** [review]
LGTM

File: Sources/Swift/Helper/SentryCurrentDateProvider.swift (hunk 1 above)
Comment (philipphofmann) [reply Sources/Swift/Helper/SentryCurrentDateProvider.swift:1]: Yes we do, we need it because of `getAbsoluteTime()`, which is in `SentryTime.h`

**Pull request merged**